### ISSUE TITLE
How can we incorporate the choice of which data (test/train/both) of a task to use for calculation of a measure?

### ISSUE THREAD

**Issue by bblodfon** [mlr-org/mlr3#1333]
So now when we do:

```
re = resample(task, learner, resampling)
re$score(msr("surv.brier"))
```

and the measure takes the arguments `task` and `train_set` as [in here](https://github.com/mlr-org/mlr3proba/blob/main/R/MeasureSurvGraf.R#L88), the `.score` function will have access to the training dataset to perform some estimation, eg in the survival analysis usually we estimate the censoring distribution via Kaplan-Meier, $G(t)$. In the non-resampling case, if the `train` and `train_set` are not used, the test data will be used for such purposes.

We now have evidence that the choice of data (train / test / both) that is used to calculate $G(t)$ [paper link](https://pubmed.ncbi.nlm.nih.gov/39888901/) can influence positively or negatively the score, so it would be nice to have a more general way to say "apply this score and estimate some quantites that are required by the score using only the test set or train set or both" during resampling (and non-resampling) schemes. Note that this is not related to which observations the score is calculated for (use of `predict_sets`).

**Comment by sebffischer** [comment]
This is already possible by specifying the `$predict_sets` field of `Measure`:

``` r
mlr3::msr("regr.mse", predict_sets = "train")
#> 
#> ── <MeasureRegrSimple> (regr.mse): Mean Squared Error ──────────────────────────
#> • Packages: mlr3 and mlr3measures
#> • Range: [0, Inf]
#> • Minimize: TRUE
#> • Average: macro
#> • Parameters: list()
#> • Properties: weights
#> • Predict type: response
#> • Predict sets: train
#> • Aggregator: mean()
```

The new book chapter also covers this in depth: https://mlr3book.mlr-org.com/chapters/chapter15/predsets_valid_inttune.html#sec-predict-sets

**Comment by bblodfon** [comment]
Sebastian I didn't explain this well, I re-phrased it now in the 1st post. It's different from what I understand than the "which observations is the score applied to" which is the `predict_sets` argument you mention.

**Comment by bblodfon** [comment]
Discussed in workshop: as this is a very specific use case we won't implement this globally in `mlr3`. An extra parameter `cens_obs` = `"all"`|`"train"`|`"test"` could be implemented for the scores like `surv.brier` that use it.
